pkg/result/processors/sort_results.go

```diff
@@ -13,41 +16,67 @@ import (
 // by sorting results.Issues using processor step, and chain based
 // rules that can compare different properties of the Issues struct.
 
+const (
+	fileOrderName      = "file"
+	linterOrderName    = "linter"
+	linterSeverityName = "severity"
+)
+
 var _ Processor = (*SortResults)(nil)
 
 type SortResults struct {
-	cmp comparator
-	cfg *config.Config
+	cmps map[string][]comparator
+
+	cfg *config.Output
 }
 
 func NewSortResults(cfg *config.Config) *SortResults {
-	// For sorting we are comparing (in next order): file names, line numbers,
-	// position, and finally - giving up.
 	return &SortResults{
-		cmp: ByName{
-			next: ByLine{
-				next: ByColumn{},
-			},
+		cmps: map[string][]comparator{
+			// For sorting we are comparing (in next order):
+			// file names, line numbers, position, and finally - giving up.
+			fileOrderName: {&byName{}, &byLine{}, &byColumn{}},
+			// For sorting we are comparing: linter name
+			linterOrderName: {&byLinter{}},
```

Comment:
If you specify by linter in your search order, don't you want the groups to be sorted same as file order within those groups?

```sh
file1.go:1 (linter-a)
file2.go:2 (linter-a)
file1.go:2 (linter-b)
```

Rather than
```sh
file2.go:2 (linter-a)
file1.go:1 (linter-a)
file1.go:2 (linter-b)
```

Or would you just require the end user to specify both in the config?

Reply:
you have to define the order you want, because you can want `file > linter` or `linter > file`